src/core/timer.c

```diff
@@ -332,7 +332,13 @@ static void add_random(Timer *t, usec_t *v) {
         if (*v == USEC_INFINITY)
                 return;
 
-        add = random_u64() % t->random_usec;
+        if (t->randomize_first_run_only)
+                if (!dual_timestamp_is_set(&t->last_trigger))
```

Comment:
why bind this to last_trigger btw? wouldn't it make more sense to just check if `t->random_value_usec` is `!= USEC_INFINITY`? i.e. initialize the field to USEC_INFINITY as a marker for "has not been set yet".

Reply:
also, this value should be serialized, i.e. timer_serialize() needs a line to write the current value out, and timer_deserialize_item() needs to be able to parse it again.

Reply:
The USEC_INFINITY is definitely a better option.
Also completely forgot about (de)serialization, thanks for reporting!